protocols/v2/noise-sv2/src/initiator.rs

```diff
@@ -226,25 +230,47 @@ impl Initiator {
             .0
             .serialize();
         let rs_pk_xonly = XOnlyPublicKey::from_slice(&rs_pub_key).unwrap();
-        if signature_message.verify(&rs_pk_xonly) {
-            let (temp_k1, temp_k2) = Self::hkdf_2(self.get_ck(), &[]);
-            let c1 = ChaCha20Poly1305::new(&temp_k1.into());
-            let c2 = ChaCha20Poly1305::new(&temp_k2.into());
-            let c1: Cipher<ChaCha20Poly1305> = Cipher::from_key_and_cipher(temp_k1, c1);
-            let c2: Cipher<ChaCha20Poly1305> = Cipher::from_key_and_cipher(temp_k2, c2);
-            self.c1 = None;
-            self.c2 = None;
-            let mut encryptor = GenericCipher::ChaCha20Poly1305(c1);
-            let mut decryptor = GenericCipher::ChaCha20Poly1305(c2);
-            encryptor.erase_k();
-            decryptor.erase_k();
-            let codec = crate::NoiseCodec {
-                encryptor,
-                decryptor,
-            };
-            Ok(codec)
-        } else {
-            Err(Error::InvalidCertificate(plaintext))
+        match &self.responder_authority_pk {
+            Some(responder_authority_pk) => {
+                if signature_message.verify(&rs_pk_xonly, responder_authority_pk) {
```

Comment:
I think that it would be better to move the match inside the `.verify()` method, uin order to have a cleaner step2 code. Something like
`
    pub fn verify(self, pk_: &Option<XOnlyPublicKey>) -> bool {
        match pk_ {
            Some(pk) => {
                let now = SystemTime::now()
                    .duration_since(SystemTime::UNIX_EPOCH)
                    .unwrap()
                    .as_secs() as u32;
                if self.valid_from <= now && self.not_valid_after >= now {
                    let secp = Secp256k1::verification_only();
                    let (m, s) = self.split();
                    let m = Message::from_hashed_data::<sha256::Hash>(&m[0..10]);
                    let s = match Signature::from_slice(&s) {
                        Ok(s) => s,
                        _ => return false,
                    };
                    secp.verify_schnorr(&s, &m, pk).is_ok()
                } else {
                    false
                }
            },
            None => true
        }
    }
`

Reply:
done https://github.com/stratum-mining/stratum/commit/2591a9bd2a3f47bbe27ac2ddf3fd9b13f42cfc85